### PR TITLE
Added optional value in first and last

### PR DESCRIPTION
**Pull request by yasserhussain1110**

### CODE REVIEW

**Comment by rgbkrk** [comment]
Looks solid to me!

File: __tests__/Seq.ts [hunk 1]
```diff
@@ -10,6 +10,14 @@
 import { isCollection, isIndexed, Seq } from '../';
 
 describe('Seq', () => {
+  it('has a default first value', () => {
+    expect(Seq().first({})).toEqual({});
+  });
+
+  it('has a default first value', () => {
+    expect(Seq().last({})).toEqual({});
+  });
```
Comment (rgbkrk) [inline __tests__/Seq.ts]: Should probably also have a test that ensures that calling `first` and `last` on an empty `Seq` with no default set will return `undefined`. Only to make sure the behavior is consistent with the past.

**Comment by yasserhussain1110** [comment]
@rgbkrk Made the required changes.

**Comment by asazernik** [comment]
Add this optional argument to the type definitions (`type-definitions/Immutable.d.ts` and `type-definitions/immutable.js.flow`) and describe it in the docs (the javadoc-ish comments in `type-definitions/Immutable.d.ts`).

**Comment by yasserhussain1110** [comment]
Added type definitions. Please let me know if further changes are required.

**Comment by rgbkrk** [comment]
@flarnie @leebyron -- this PR has been put through the paces -- would you be willing to check it out for incorporation into Immutable 4.0?

**Comment by hrldcpr** [comment]
Note that this PR leads to new type errors since `NSV` defaults to `{}`—see this comment for example https://github.com/facebook/immutable-js/commit/37ae5beef11664371ecb231b422a25ffa11126eb#r31116027
